homework/src/index.js

```diff
@@ -31,17 +33,116 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
+    const root = document.getElementById("root");
+
+    const div = createAndAppend("div", root);
+    const header = createAndAppend("p", div, {
+      text: "HYF Repositories",
+      class: "header"
     });
+
+    const table = createAndAppend("table", root, { class: "table" });
+    const select = createAndAppend("select", header, { class: "select" });
+
+    const tbody = createAndAppend("tbody", table);
+    const tr = createAndAppend("tr", tbody);
+    const container = createAndAppend("div", root, { class: "container" });
+    createAndAppend("td", tr, { text: "Repository: ", class: "label" });
+    const repositoryLink = createAndAppend("a", tbody);
+    const description = createAndAppend("tr", tbody);
+    const forks = createAndAppend("tr", tbody, { class: "label" });
+
+    const updated = createAndAppend("tr", tbody, { class: "label" });
+
+    fetchJSON(url)
+      .catch(error => {
```

Comment:
catch or then first ? I think it's very important question and here is the [answer](https://youtu.be/M2R2pkKv1k8?t=1054) I think you watched it before but re-watching things we did watched them before is important after applying it sometimes :slightly_smiling_face: